aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -0,0 +1,132 @@
+#include "ATen/native/cpu/UnaryOpsKernel.h"
+#include "ATen/Dispatch.h"
+#include "ATen/Parallel.h"
+#include <iostream>
+#include <cmath>
+#include "ATen/native/cpu/Vec256.h"
+
+namespace at {
+namespace native {
+
+using namespace vec256;
+
+// This modifies arr in place with given OP
+template <class scalar_t, template <class> class VOP, template <class> class SOP, CPUCapability C>
+inline void kernel_(scalar_t *arr_out, const scalar_t *arr_in, size_t start, size_t end) {
+  // Use all 16 registers.
+  Vec256<scalar_t> a[8];
+  size_t width =
+      256 / sizeof(scalar_t); // primitives per 256 bytes (two cache lines)
+  size_t epr = 32 / sizeof(scalar_t); // primitives per Vec256
+  size_t k = 0;
+  for (; k < (end - start) / width; k++) {
+    for (size_t i = 0; i < 8; i++) {
+      a[i].load(arr_in + (k * width) + i * epr + start);
+    }
+    for (size_t i = 0; i < 8; i++) {
+      VOP<scalar_t>()(a[i]);
+    }
+    for (size_t i = 0; i < 8; i++) {
+      a[i].store(arr_out + (k * width) + i * epr + start);
+    }
+  }
+  k = k * width + start;
+  for (; k < end; k++) {
+    arr_out[k] = SOP<scalar_t>()(arr_in[k]);
```

Comment:
This can be simplified using partial loads and stores as suggested by Sam Gross.